File: function-aws/features/test-aws-junit/skeleton/src/test/java/@defaultPackage.path@/@project.className@Client.java
package @defaultPackage@;

import io.micronaut.function.client.FunctionClient;
import io.micronaut.http.annotation.Body;
import io.reactivex.Single;
import javax.inject.Named;

@FunctionClient
public interface @project.className@Client {

    @Named("@project.name@")
    Single<@project.className@> apply(@Body @project.className@ body);

}

File: function-aws/features/test-aws-spek/skeleton/src/test/kotlin/@defaultPackage.path@/@project.className@Client.kt
package @defaultPackage@

import io.micronaut.function.client.FunctionClient
import io.micronaut.http.annotation.Body
import io.reactivex.Single
import javax.inject.Named

@FunctionClient
interface @project.className@Client {

    @Named("@project.name@")
    fun apply(@Body body : @project.className@): Single<@project.className@>

}
